1.3/spec.md

```diff
@@ -0,0 +1,180 @@
+implementation details of grids and interaction metadata itself: the MBTiles
+specification is only concerned with storage._
+
+#### Schema
+
+The database can have optional tables named `grids`, `grid_data`.
+
+The `grids` table must yield four columns named `zoom_level`, `tile_column`,
+`tile_row`, and `grid`. A typical create statement for the `grids` table:
+
+    CREATE TABLE grids (zoom_level integer, tile_column integer, tile_row integer, grid blob);
+
+The `grid_data` table must yield five columns named `zoom_level`, `tile_column`,
+`tile_row`, `key_name`, and `key_json`. A typical create statement for the `grid_data` table:
+
+    CREATE TABLE grid_data (zoom_level integer, tile_column integer, tile_row integer, key_name text, key_json text);
+
+#### Content
+
+The `grids` table contains UTFGrid data, gzip compressed.
+
+The `grid_data` table contains grid key to value mappings, with values encoded
+as JSON objects.
+
+## Vector tileset metadata
+
+As mentioned above, Mapbox Vector Tile tilesets must include a `json` row in the `metadata` table
+to summarize what layers are available in the tiles and what attributes are available for the
+features in those layers.
+
+The row contains a single JSON object with a `vector_layers` key, whose value is an array of layers.
+Each layer is a JSON object with the following keys:
+
+* `id`: The layer ID, used for a [CartoCSS selector](https://tilemill-project.github.io/tilemill/docs/guides/selectors/) or [Mapbox GL Style layer](https://www.mapbox.com/mapbox-gl-style-spec/#layers).
+* `description`: A human-readable description of the layer's contents (or empty string if not available).
+* `fields`: A JSON object whose keys and values are the names and types of attributes available in this layer. The type should be `Number`, `Boolean`, or `String`.
+
+The layer object may also contain these keys:
+
+* `minzoom`: The lowest zoom level whose tiles this layer appears in.
+* `maxzoom`: The highest zoom level whose tiles this layer appears in.
+
+### Example
+
+A vector tileset that contains United States counties and primary roads from TIGER might
+have the following metadata table:
+
+* `name`: `TIGER 2016`
+* `format`: `mvt`
+* `bounds`: `-179.231086,-14.601813,179.859681,71.441059`
+* `center`: `-84.375000,36.466030,5`
+* `minzoom`: `0`
+* `maxzoom`: `5`
+* `attribution`: `United States Census`
+* `description`: `US Census counties and primary roads`
+* `type`: `overlay`
+* `version`: `2`
+* `scheme`: `tms`
+* `json`: `{"vector_layers": [ { "id": "tl_2016_us_county", "description": "Census counties", "minzoom": 0, "maxzoom": 5, "fields": {"ALAND": "Number", "AWATER": "Number", "COUNTYFP": "String", "GEOID": "String", "MTFCC": "String", "NAME": "String", "NAMELSAD": "String", "STATEFP": "String"} }, { "id": "tl_2016_us_primaryroads", "description": "Census primary roads", "minzoom": 0, "maxzoom": 5, "fields": {"FULLNAME": "String", "LINEARID": "String", "MTFCC": "String", "RTTYP": "String"} } ] }`
```

Comment:
Since this is a pseudo-table anyway, maybe pretty format the `json` field for legibility?